virttest/qemu_vm.py

```diff
@@ -4852,7 +4852,7 @@ def deactivate_netdev(self, nic_index_or_name):
         if not verify:
             LOG.error(self.monitor.info("network", debug=False))
             raise virt_vm.VMDelNetDevError("Fail to remove netdev %s" % netdev_id)
-        if "tapfd_ids" in nic:
+        if "tapfd_ids" in nic and "tapfds" in nic:
```

Comment:
See this: https://github.com/avocado-framework/avocado-vt/blob/3bd517aa16235eae62fa89c1ff84f027592d6b5c/virttest/qemu_vm.py#L4704-L4716

nic.tapfds is also not in the hotplugged NIC device, so this change will break the original design.

Discussed with @YongxueHong, we think you can have a try of qmp command `query-fdsets`, hope it can check if each tapfd_id in the output before colseing it.

Reply:
As I know, nic.tapfds is in the hotplugged NIC device. For query-fdsets, seems need add fd to a fd set before query-fdsets,but framwork didn't do that for original nic.So I can't check tapfd_id  directly with query-fdsets here.

Reply:
Right, sorry I missed https://github.com/avocado-framework/avocado-vt/blob/c009cee69a6425f58bf644103d3150b883c49834/virttest/qemu_vm.py#L4740

If fdsets doesn't work, another suggestion is del tapfd_ids together during del tapfds, as they are combined. https://github.com/avocado-framework/avocado-vt/blob/c009cee69a6425f58bf644103d3150b883c49834/virttest/qemu_vm.py#L3836-L3856

But, both methods should be fine. #YongxueHong FYI